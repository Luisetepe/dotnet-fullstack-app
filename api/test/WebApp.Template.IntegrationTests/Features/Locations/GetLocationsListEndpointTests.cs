using Snapshooter;
using Snapshooter.Xunit;
using WebApp.Template.Application.Features.Locations.Queries.GetLocationsList;
using WebApp.Template.Endpoints.Locations.GetLocationsList;

namespace WebApp.Template.IntegrationTests.Features.Locations;

public class GetLocationsListEndpointTests(IntegrationTestFixture fixture)
    : TestBase<IntegrationTestFixture>
{
    [Theory]
    [InlineData(1, 5, "name", "asc", "Loc")]
    [InlineData(1, 5, "name", "desc", "")]
    [InlineData(1, 5, "latitude", "asc", "cat")]
    [InlineData(1, 5, "latitude", "desc", "5")]
    [InlineData(1, 5, "longitude", "asc", "on 2")]
    [InlineData(1, 5, "longitude", "desc", "3")]
    public async Task Should_Return_Locations(
        int page,
        int pageSize,
        string sortBy,
        string sortDirection,
        string search
    )
    {
        // Act
        var (httpResponse, result) = await fixture.Client.GETAsync<
            GetLocationsListEndpoint,
            GetLocationsListRequest,
            GetLocationsListResponse
        >(
            new GetLocationsListRequest
            {
                PageNumber = page,
                PageSize = pageSize,
                OrderBy = sortBy,
                Order = sortDirection,
                Search = search
            }
        );

        // Assert
        httpResponse.StatusCode.Should().Be(HttpStatusCode.OK);

        result.Should().NotBeNull();
        result
            .Should()
            .MatchSnapshot(
                SnapshotNameExtension.Create(page, pageSize, sortBy, sortDirection, search),
                matchOptions => matchOptions.IgnoreFields("**.Id")
            );
    }
}
