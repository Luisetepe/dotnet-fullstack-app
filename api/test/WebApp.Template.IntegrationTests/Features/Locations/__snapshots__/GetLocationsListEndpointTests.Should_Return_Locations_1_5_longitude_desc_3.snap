{
  "Locations": [
    {
      "Id": "0G5GJEPTMF4G6",
      "Name": "Location 3",
      "Longitude": -179.987654,
      "Latitude": 89.987654
    }
  ],
  "Pagination": {
    "CurrentPageNumber": 1,
    "CurrentPageSize": 5,
    "TotalRows": 1,
    "TotalPages": 1
  }
}
